homeassistant/components/input_select/__init__.py

```diff
@@ -162,12 +232,17 @@ def should_poll(self):
     @property
     def name(self):
         """Return the name of the select input."""
-        return self._name
+        return self._config.get(CONF_NAME)
 
     @property
     def icon(self):
         """Return the icon to be used for this entity."""
-        return self._icon
+        return self._config.get(CONF_ICON)
+
+    @property
+    def _options(self) -> typing.List[str]:
```

Comment:
This is not used.

Reply:
I contemplated replacing references to it in https://github.com/home-assistant/home-assistant/blob/d62a744c14d1c70c81431db5d6958d961e45bad3/homeassistant/components/input_select/__init__.py#L264, https://github.com/home-assistant/home-assistant/blob/d62a744c14d1c70c81431db5d6958d961e45bad3/homeassistant/components/input_select/__init__.py#L276, https://github.com/home-assistant/home-assistant/blob/d62a744c14d1c70c81431db5d6958d961e45bad3/homeassistant/components/input_select/__init__.py#L283 with `self._config[CONF_OPTIONS]` but it is less readable so decided against  it.